src/KDSoapClient/KDSoapMessageAddressingProperties.cpp

```diff
@@ -217,16 +217,34 @@ KDSoapMessageAddressingProperties::~KDSoapMessageAddressingProperties()
 
 QString KDSoapMessageAddressingProperties::predefinedAddressToString(KDSoapMessageAddressingProperties::KDSoapAddressingPredefinedAddress address, KDSoapAddressingNamespace addressingNamespace)
 {
-    const QString addressingNS = addressingNamespaceToString(addressingNamespace);
+    QString prefix = addressingNamespaceToString(addressingNamespace);
+    // Up to and including the 2004/08 spec the well-known URIs have a /role/
+    // folder in their path and furthermore only /anonymous is actually
+    // well-known.
+    switch (addressingNamespace) {
+    case Addressing200303:
+    case Addressing200403:
+    case Addressing200408: {
+        prefix += QLatin1String("/role");
+        if (!address == Anonymous) {
```

Comment:
Err, did you mean `if (address != Anonymous)` ?

Reply:
Oh my, yes. Fixed